taipy/gui/gui.py

```diff
@@ -2411,19 +2414,24 @@ def _notify(
         system_notification: t.Optional[bool] = None,
         duration: t.Optional[int] = None,
         notification_id: t.Optional[str] = None,
+        on_close: t.Optional[t.Callable[[State, str, str], None]] = None,
     ):
+        if notification_id and on_close:
+            self._notification_callbacks[notification_id] = on_close
```

Comment:
`on_close` should be transformed into a string so that it can be managed by the front-end. (check the [builder code](https://github.com/Avaiga/taipy/blob/5c56f125a2bab02a260eee88503ee480ac933f7e/taipy/gui/_renderers/builder.py#L445) for inspiration)
so that there is a dispatch from the front-end to the back-end only if `on_close` is defined.